torch/_subclasses/meta_utils.py

```diff
@@ -551,6 +555,15 @@ def __call__(
             # tensors into their trace / some subclasses don't correctly
             # support meta.  Trying to YOLO this is more trouble than it's
             # worth.
+            if is_traceable_wrapper_subclass(t):
+                out = transform_subclass(
+                    t,
+                    lambda t: self.meta_tensor(
+                        t, shape_env=shape_env, callback=callback, source=source
+                    ),
+                )
+                return out
```

Comment:
Unfortunately I don't think this works :(. In particular, in my AOTDispatch prototype, I realized that fake-ifying this way causes your subclass to lose autograd metadata (the original `DTensor` has `requires_grad=True`, the updated one has `requires_grad=False`).

@wanchaol and I talked a bit about this offline. I have an updated version that seems to preserve autograd metadata in my prototype here: https://github.com/pytorch/pytorch/pull/104483/files#diff-ae6c64c5b6b4fbfa293a1ce7e82d4a9b9b7ee7cdca8ef66312031b52c69a0b4aR155. This version works on my basic examples, but fails for `DTensor` (because the meta conversion logic ends up calling `.set_()`, which `DTensor` sees and can't handle).

Reply:
Yeah we need to look into what this `meta_tensor` did for the `set_` call and see how that applies to DTensor. I once understood how that functions works but now my memory got damaged on it.. I'll re-read this and see what's going on there :)

Reply:
discussed offline, @bdhirsh is going to make this sound in https://github.com/pytorch/pytorch/pull/104482